faas/lambda_function.py

```diff
@@ -0,0 +1,43 @@
+"""
+This the handler invoked by the Lambda function that runs whenever there's a change to a hub's uploaded model-output files.
+For the Hubverse AWS account, the baseline "hubverse-transform-model-output" Lambda is defined via our IaC (Infrastructure as Code)
+repository: hubverse-transform-model-output.
+
+This handler, along with the actual transformation module (hubverse_transform), live outside of the Hubverse's IaC repository:
+
+- To avoid tightly coupling AWS infrastructure to the more general hubverse_transform module that can be used for hubs hosted elsewhere
+- To allow faster iteration and testing of the hubverse_transform module without needing to update the IaC repo or redeploy AWS resources
+"""
+import json
+import logging
+import urllib.parse
+
+from hubverse_transform.model_output import ModelOutputHandler
+
+logger = logging.getLogger()
+logger.setLevel("INFO")
+
+
+def lambda_handler(event, context):
```

Comment:
The purpose of this function is to receive the AWS S3 events that are emitted whenever a new/updated `model-output` file lands in a hub's S3 bucket. The function parses the event to get the name of the hub's S3 bucket and the name ("key") of the model-output file.

The bucket name + key are then used to create a`ModelOutputHandler` object and transform the data.

For reference, this is an example of such an event
```
{
    "Records": [
        {
            "eventVersion": "2.1",
            "eventSource": "aws:s3",
            "awsRegion": "us-east-1",
            "eventTime": "2024-05-02T19:06:28.151Z",
            "eventName": "ObjectCreated:Put",
            "userIdentity": {
                "principalId": "howdy"
            },
            "requestParameters": {
                "sourceIPAddress": "redacted"
            },
            "responseElements": {
                "x-amz-request-id": "S7DY1N8KZP1F8J35",
                "x-amz-id-2": "LeRQgYUNlXYiMdY+ibFpvF0XUSjcu5tgyUyhufmsSavl+oPrpKF2L5/J1MYfe+F0wEKHUGnC+BxxOrQKhTiXPKmuAxVFA48R"
            },
            "s3": {
                "s3SchemaVersion": "1.0",
                "configurationId": "howdy",
                "bucket": {
                    "name": "hubverse-cloud",
                    "ownerIdentity": {
                        "principalId": "howdy"
                    },
                    "arn": "arn:aws:s3:::hubverse-cloud"
                },
                "object": {
                    "key": "raw/model-output/UMass-flusion/2023-10-14-UMass-flusion.csv",
                    "size": 357048,
                    "eTag": "af50008c99f29b39310b57be5f474d28",
                    "versionId": "nPvSV7RK1UNFXMGHNY9pDiNVbb5zBTJI",
                    "sequencer": "006633E433E70A377B"
                }
            }
        }
    ]
}
```